src/charm.py

```diff
@@ -1054,59 +1054,45 @@ def _configure_pki_secrets_engine(self) -> None:
         if not self._common_name_config_is_valid():
             logger.debug("Common name config is not valid, skipping")
             return
-        common_name = self._get_config_common_name()
-        vault.enable_secrets_engine(SecretsBackend.PKI, VAULT_PKI_MOUNT)
-        if not self._is_intermediate_ca_common_name_valid(vault, common_name):
-            csr = vault.generate_pki_intermediate_ca_csr(
-                mount=VAULT_PKI_MOUNT, common_name=common_name
-            )
-            self.tls_certificates_pki.request_certificate_creation(
-                certificate_signing_request=csr.encode(),
-                is_ca=True,
-            )
-            self._set_pki_csr_secret(csr)
-
-    def _is_intermediate_ca_common_name_valid(self, vault: Vault, common_name: str) -> bool:
-        """Check if the intermediate CA is set with the valid common name."""
-        intermediate_ca = vault.get_intermediate_ca(mount=VAULT_PKI_MOUNT)
-        if not intermediate_ca:
-            return False
-        intermediate_ca_common_name = get_common_name_from_certificate(intermediate_ca)
-        return intermediate_ca_common_name == common_name
-
-    def _is_intermediate_ca_set(self, vault: Vault, certificate: str) -> bool:
-        """Check if the intermediate CA is set in the PKI secrets engine."""
-        intermediate_ca = vault.get_intermediate_ca(mount=VAULT_PKI_MOUNT)
-        return certificate == intermediate_ca
-
-    def _add_intermediate_ca_certificate_to_pki_secrets_engine(self) -> None:
-        """Add the CA certificate to the PKI secrets engine."""
-        if not self.unit.is_leader():
-            logger.debug("Only leader unit can handle a vault-pki certificate request")
+        config_common_name = self._get_config_common_name()
+        if not config_common_name:
+            logger.error("Common name is not set in the charm config")
             return
-        vault = self._get_active_vault_client()
-        if not vault:
-            logger.debug("Vault is not ready to handle a vault-pki certificate request")
+        certificate_request = self._get_certificate_request()
+        if not certificate_request:
+            logger.error("Certificate request is not valid")
             return
-        certificate = self._get_pki_intermediate_ca_certificate()
-        if not certificate:
+        provider_certificate, private_key = self.tls_certificates_pki.get_assigned_certificate(
```

Comment:
`provider_certificate` is somewhat confusing to me because of the word "provider" and the meaning that word can carry in a charm, how about?
```suggestion
        intermediate_ca_certificate, private_key = self.tls_certificates_pki.get_assigned_certificate(
```

Reply:
I mean it is the certificate that the provider of the TLS certificates integration generated. Plus I wish we got those comments in the vault k8s charm because everything I adapt here will mean that the machine charm would be different from the k8s one.

Reply:
We can keep it. For me it's difficult to read because while it is true that this is a ProviderCertificate object coming from a provider, PKI itself is a provider and the name `provider_certificate` is used in the context of PKI providing certificates too in the same file.

Reply:
I don't think having to make the same changes in both charms should be a limiting factor here. But I agree that it means a lot of back and forth, because if you create a new PR just to get the modification made here in k8s you might get new comments there.
I think TLSENG-120 would help :)